docs/_posts/2016-08-19-fund-12-Oth-CD.md

```diff
@@ -7,6 +7,281 @@ order: 3
+            sh './gradlew build'
+            archiveArtifacts artifacts: '**/ordering-service*.jar', excludes: null
+            junit '**/TEST-*.xml'
+        }
+    }
+
+    stage ('Build Client') {
+        dir('src/Clients') {
+            sh 'chmod +x gradlew'
+            sh './gradlew build'
+            archiveArtifacts artifacts: '**/mrp.war', excludes: null
+        }
+    }
+    
+    stage ('Save MongoRecords.js') {
+        dir('deploy') {
+            archiveArtifacts artifacts: 'MongoRecords.js', excludes: '**/docker'
+        }
+    }
+
+    def mrphostname
+    
+    stage ('Get MRP hostname') { 
+        mrphostname = input(
+            id: 'mrhphostname', message: 'Please enter the MRP VM FQDN', parameters: [
+                [$class: 'TextParameterDefinition', defaultValue: 'mrp-deploy.westus.cloudapp.azure.com', description: 'This is the FQDN by which the VM can be remotely accessed', name: 'mrphostname']
+                ])
+        echo ("MRP HostName: ${mrphostname}")
+    }
+    
+    stage ("ssh") {
+        sshagent(['mrpadmin']) {
+            sh "ssh -o StrictHostKeyChecking=no -l mrpadmin  ${mrphostname} 'curl -O ${env.JENKINS_URL}/job/PartsUnlimitedMRP/lastSuccessfulBuild/artifact/build/libs/mrp.war'"
+            sh "ssh -o StrictHostKeyChecking=no -l mrpadmin  ${mrphostname} 'curl -O ${env.JENKINS_URL}/job/PartsUnlimitedMRP/lastSuccessfulBuild/artifact/build/libs/integration-service-0.1.0.jar'"
+            sh "ssh -o StrictHostKeyChecking=no -l mrpadmin  ${mrphostname} 'curl -O ${env.JENKINS_URL}/job/PartsUnlimitedMRP/lastSuccessfulBuild/artifact/build/libs/ordering-service-0.1.0.jar'"
+            sh "ssh -o StrictHostKeyChecking=no -l mrpadmin  ${mrphostname} 'curl -O ${env.JENKINS_URL}/job/PartsUnlimitedMRP/lastSuccessfulBuild/artifact/MongoRecords.js'"
+            sh "ssh -o StrictHostKeyChecking=no -l mrpadmin  ${mrphostname} 'curl -sL https://raw.githubusercontent.com/dcaro/PartsUnlimitedMRP/master/docs/assets/jenkins/env/deploy_mrp_app.sh | sudo bash -'"
+        }
+        sh "curl -sL -w \"%{http_code}\\n\" http://${mrphostname}:9080/mrp/ -o /dev/null"
+    }
+    
+}
+
+{% endhighlight %}
+
+
+### Taks 4: Connecting the dots
+
+**1.** Go to the source code on your github repository
+
+**2.** Edit the file **src/Clients/Web/index.html**
+
+You can change the title of the page for example.
+
+![Editing PartsUnlimitedMRP title](<../assets/jenkins/edit_index.png>)
+
+**3.** Click **Commit changes** 
+
+**4.** Navigate to your Jenkins master
+```
+http://fqdn_of_your_jenkinsmaster:8080/job/PartsUnlimitedMRP/
+```
+
+**5.** If not already there, the pipeline will stop at the stage **Get MRP hostname**
+
+Click on the step that is paused and enter the FQDN of the machine that you have deployed in Task 2.3.
+
+Click **Proceed** 
+
+![Enter the FQDN of the deployment machine](<../assets/jenkins/pipeline_wait_for_userinput.png>)
+
+**6.** Wait until the completion of the pipeline 
+
```

Comment:
This is because you did not type the hostname in the first lab, the variable ${evn.JENKINS_URL} will return null in this case and the mongoDb will not be populated.